nx_parallel/utils/chunk.py

```diff
@@ -2,65 +2,96 @@
 import os
 import networkx as nx
 
+from nx_parallel.utils.types import GraphIteratorType
+
+
 __all__ = ["chunks", "get_n_jobs", "create_iterables"]
 
 
-def chunks(iterable, n):
-    """Divides an iterable into chunks of size n"""
+def chunks(iterable, n_chunks):
+    """Yield exactly `n_chunks` chunks from `iterable`, balancing the chunk sizes."""
+    iterable = list(iterable)
+    k, m = divmod(len(iterable), n_chunks)
     it = iter(iterable)
-    while True:
-        x = tuple(itertools.islice(it, n))
-        if not x:
-            return
-        yield x
+    for _ in range(n_chunks):
+        chunk_size = k + (1 if m > 0 else 0)
+        m -= 1
+        yield tuple(itertools.islice(it, chunk_size))
 
 
 def get_n_jobs(n_jobs=None):
-    """Returns the positive value of `n_jobs`."""
+    """
+    Return the positive value of `n_jobs`, adjusting for the environment.
+
+    If running under pytest, returns 2 jobs. If using a parallel backend,
+    returns the number of jobs configured for the backend. Otherwise, returns
+    the number of CPUs adjusted for negative `n_jobs` values.
+    """
     if "PYTEST_CURRENT_TEST" in os.environ:
         return 2
-    else:
+
+    if n_jobs == 0:
+        raise ValueError("n_jobs == 0 in Parallel has no meaning")
+
+    if not n_jobs:
         if nx.config.backends.parallel.active:
             n_jobs = nx.config.backends.parallel.n_jobs
         else:
             from joblib.parallel import get_active_backend
 
             n_jobs = get_active_backend()[1]
-        n_cpus = os.cpu_count()
-        if n_jobs is None:
-            return 1
-        if n_jobs < 0:
-            return n_cpus + n_jobs + 1
-        if n_jobs == 0:
-            raise ValueError("n_jobs == 0 in Parallel has no meaning")
-        return int(n_jobs)
-
-
-def create_iterables(G, iterator, n_cores, list_of_iterator=None):
-    """Creates an iterable of function inputs for parallel computation
+
+    if not n_jobs:
+        return 1  # Default to 1 if no valid n_jobs is found or passed
+    if n_jobs < 0:
+        return os.cpu_count() + n_jobs + 1
+
+    return int(n_jobs)
+
+
+def create_iterables(
+    G,
+    iterator,
+    n_cores,
+    list_of_iterator=None,
+):
+    """
+    Create an iterable of function inputs for parallel computation
     based on the provided iterator type.
 
     Parameters
-    -----------
-    G : NetworkX graph
-    iterator : str
-        Type of iterator. Valid values are 'node', 'edge', 'isolate'
-
-    Returns:
-    --------
-    iterable : Iterable
+    ----------
+    G : networkx.Graph
+        The NetworkX graph.
+    iterator : GraphIteratorType
+        Type of iterator. Valid values are 'NODE', 'EDGE', 'ISOLATE'.
```

Comment:
The process of creating the enum GraphIteratorType makes this much harder to read -- involving looking across modules only to discover that they are just strings. Let's just use strings. Is there a reason for the extra complexity? If you have ideas of how this might grow to become worth the effort then can you explain that expectation?  The confusion between lowercase values and upper case attribute names seems like a C idiom, but could be confusing to some python users.

Also, while I believe the user could use strings as the inputs, that is not at all clear from the docs. If we stay with the enum approach then let's make these docs set up for string input.

Finally, should these be plural instead of singular? As in "iterator over nodes" or "iterator yields nodes". Not a blocker of course, but maybe we should think about it.

Reply:
Using enums could be worth the trade-off for the type safety / clarity. They help prevent errors from typos in string literals, ensure consistency, and make the code more self-documenting and extendable for future iterations. That being said, I definitely see your point about keeping things simple and reducing overhead, especially if the set of iterator types isn't expected to grow. One potential compromise would be to allow both enums and strings, which could provide the best of both worlds—type safety when desired, and simplicity for users who prefer the string-based approach. 

Regarding the suggestion about plural vs. singular names (e.g., "nodes" instead of "node"), I think that's a great point. Using the plural form seems more intuitive since it better reflects what the iterator actually returns—multiple nodes, edges, or isolates. I've updated the iterator type names to be plural for consistency and clarity.

Like with the type annotations, I don't feel strongly about enum handling so am also happy to remove the enums for this PR altogether if it would expedite review. WDYT?

Reply:
Sounds good, removed and reverted for now